src/App.tsx

```diff
@@ -1,222 +1,273 @@
+  };
+
+  const clearCompleted = () => {
+    const newTodos = todos.filter(todo => !todo.completed);
+
+    setTodos(newTodos);
+  };
+
+  const changePosition = (newPosition: string) => {
+    setPosition(newPosition);
+  };
+
+  const handleNewTodo = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const userTodo = {
+      id: currTodoId,
+      userId: user?.id,
+      title: event.target.value,
+      completed: false,
+    };
+
+    setCurrTodoId(current => current + 1);
+    setNewTodo(userTodo);
+  };
+
+  const closeErrorMessage = () => {
+    setHidden(true);
+  };
+
+  const handleTodoChange = (
+    event: React.ChangeEvent<HTMLInputElement>, todoId: number,
+  ) => {
+    const newTodos = todos.map(todo => {
+      if (todo.id !== todoId) {
+        return todo;
+      }
+
+      return {
+        ...todo,
+        title: event.target.value,
+      };
+    });
+
+    setTodos(newTodos);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (newTodo) {
+      sendNewTodo(newTodo);
+      setNewTodo(null);
+    }
+
+    newTodoField.current?.blur();
+  };
+
+  const removeTodo = (todoId: number) => {
+    deleteCurrentTodo(todoId);
+  };
+
+  const handleCurrentTodoChange = (
+    event: React.FormEvent<HTMLFormElement> | null,
+    todoId: number,
+    title: string,
+  ) => {
+    event?.preventDefault();
+
+    if (!title) {
+      deleteCurrentTodo(todoId);
+    } else {
+      modifieTodo(todoId, title);
+    }
+
+    setIsForm(false);
+  };
+
+  const handleCompleted = (todoId: number) => {
+    const newTodos = todos.map(todo => {
+      if (todo.id !== todoId) {
+        return todo;
+      }
+
+      const changedTodo = {
+        ...todo,
+        completed: !todo.completed,
+      };
+
+      return changedTodo;
+    });
+
+    setTodos(newTodos);
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    loadTodos();
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    setTimeout(() => {
+      setError('');
+      closeErrorMessage();
+    }, 3000);
+  }, [error]);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
+        <Header
+          handleSubmit={handleSubmit}
+          title={newTodo?.title || ''}
+          handleAdd={handleNewTodo}
+          newTodoField={newTodoField}
```

Comment:
```suggestion
          onFormSubmit={handleSubmit}
          title={newTodo?.title || ''}
          onAdd={handleNewTodo}
          newTodoField={newTodoField}
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
check all such cases